src/main/java/cn/nukkit/network/Network.java

```diff
@@ -75,6 +75,10 @@ public Network(Server server) {
         byte[] buf = BUFFER.get();
         while (!inflater.finished()) {
             int i = inflater.inflate(buf);
+            if (i == 0) {
+                log.debug("Prevented an infinite loop trying to decompress data. Needs input: "+inflater.needsInput()+", Needs Dictionary: "+inflater.needsDictionary());
```

Comment:
I think we should throw an exception. Needing extra input or a dictionary would suggest a corrupted zlib payload because we have the full compressed buffer here.

Reply:
Ok. I applied the suggestion.